docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,362 @@
+
+The `server` folder includes all the code for the back end of the plugin. To create it at the root of the plugin folder, run the following command in a terminal:
+
+```sh
+cd <my-plugin-folder-name>
+mkdir server
+```
+
+### Move controllers, services, and middlewares
+
+:::callout
+In Strapi v3, controllers, services, and middlewares of a plugin must follow a strict folder structure convention.
+
+In Strapi v4, the organization of files and folders for plugins is flexible. However, it is recommended to create dedicated folders for every type of back-end element (e.g. [controllers](/developer-docs/latest/developer-resources/plugin-api-reference/server.html#controllers), [services](/developer-docs/latest/developer-resources/plugin-api-reference/server.html#services), and [middlewares](/developer-docs/latest/developer-resources/plugin-api-reference/server.html#middlewares)) inside a `server` folder (see [project structure](/developer-docs/latest/setup-deployment-guides/file-structure.md)).
+
+:::
+
+To update the controllers, services, and middlewares of a plugin to Strapi v4, create specific subfolders in a `server` folder.
+
+Plugin files and folders in Strapi v4 should meet 2 requirements:
+
+- Each file in the `server/<subfolder-name>` should export a function taking the `strapi` instance (object) as a parameter and returning an object.
+- Each of the `server/<subfolder-name>` folders should include an `index.js` file that exports all files in the folder.
+
+::: details Example of files and folder for Strapi v4 plugin controllers
+
+```jsx
+// path: ./src/plugins/my-plugin/server/controllers/my-controllerA.js
+
+module.exports = ({ strapi }) => ({
+  doSomething(ctx) {
+    ctx.body = { message: "HelloWorld" };
+  },
+});
+```
+
+```jsx
+// path: ./src/plugins/my-plugin/server/controllers/index.js
+
+"use strict";
+
+const myControllerA = require("./my-controllerA");
+const myControllerB = require("./my-controllerB");
+
+module.exports = {
+  myControllerA,
+  myControllerB,
+};
+
+```
+
+:::
+
+### Move the `bootstrap` function
+
+:::callout
+Strapi v3 has a dedicated `config/functions` folder for each plugin.
+
+In Strapi v4, the `config` folder does not necessarily exist for a plugin and [the `bootstrap` function](/developer-docs/latest/developer-resources/plugin-api-reference/server.md#bootstrap) and other lifecycle functions can be declared elsewhere.
+:::
+
+To update the plugin's `bootstrap` function to Strapi v4:
+
+- move the `bootstrap()` function from `server/config/functions/bootstrap.js` to `server/bootstrap.js`
+- pass the `strapi` instance (object) as a parameter
+
+```jsx
+// path: ./src/plugins/my-plugin/server/bootstrap.js
+
+"use strict";
+
+module.exports = ({ strapi }) => ({
+  // bootstrap the plugin
+});
+```
+
+<!-- ? If we keep it this way, we should probably extend the code examples with another block, showing that strapi-server.js needs to import bootstrap.js. Or is any "bootstrap" file automatically imported? -->
```

Comment:
It's not auto-imported, default CLI generated plugins have a bootstrap file that is exported in the index.js:

![image](https://user-images.githubusercontent.com/8593673/150863047-495f6493-a4e9-4c13-a335-2becfce4f57b.png)

(There is also register and destroy)